java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -5769,6 +5769,140 @@ public void bzmpop_binary_timeout_check(BaseClient client) {
         }
     }
 
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void xadd_duplicate_entry_keys(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        String field = UUID.randomUUID().toString();
+        String foo1 = "foo1";
+        String bar1 = "bar1";
+
+        String timestamp = "1721748920942-0";
+        String[][] entry = new String[][] {{field, foo1}, {field, bar1}};
+        assertEquals(
+                timestamp,
+                client
+                        .xadd(key, entry)
+                        .get());
+
+        // get everything from the stream
+        Map<String, String[][]> result = client.xrange(key, InfRangeBound.MIN, InfRangeBound.MAX).get();
+        assertEquals(1, result.size());
+        String[][] actualEntry = result.get(timestamp);
+        assertDeepEquals(entry, actualEntry);
```

Comment:
Adding to Yury's comment, I don't think this will pass as is, but you can modify it to get it working like this:
```suggestion
        String[][] entry = new String[][] {{field, foo1}, {field, bar1}};
        String streamId =  client.xadd(key, entry).get());

        // get everything from the stream
        Map<String, String[][]> result = client.xrange(key, InfRangeBound.MIN, InfRangeBound.MAX).get();
        assertEquals(1, result.size());
        String[][] actualEntry = result.get(streamId);
        assertDeepEquals(entry, actualEntry);
```